pkg/controller/sparkapplication/controller.go

```diff
@@ -354,9 +354,15 @@ func (c *Controller) getUpdatedAppStatus(app *v1alpha1.SparkApplication) v1alpha
 			}
 		} else {
 			glog.Warningf("invalid driver state for SparkApplication %s/%s: %v", app.Namespace, app.Name, err)
+			app.Status.AppState.ErrorMessage = fmt.Sprintf("Driver Pod invalid state: %s", currentDriverState.podPhase)
+			app.Status.AppState.State = v1alpha1.FailingState
```

Comment:
invalid state here essentially means `Unknown` podPhase which can be transient from my understanding in https://kubernetes.io/docs/concepts/workloads/pods/pod-lifecycle/#pod-phase .  Currently this will be retried again after the re-sync interval which I feel is the correct behavior.

Is there a case where the pod can perpetually remain in Unknown ?